duckduckgo/macos-browser
# History in Autocomplete Suggestions (please don’t merge until BSK branch is released)

Task/Issue URL: https://app.asana.com/0/0/1200247680782237/f
Specification URL: https://app.asana.com/0/0/1200247680782237/f

**Description**:
This PR contains:

- Integration of history entries into suggestions
- Improving the ordering and scoring algorithm to achieve better user experience


**Steps to test this PR**:

Test DuckDuckGo navigational suggestions
1. If you don’t have any bookmark starting with `w`, you should get `walmart.com` as the first hit and autocompleted after typing `w` into the address bar

Test searching for a given domain
1. Type `w` into the address bar and select the third suggestion - searching for `walmart.com`
2. Make sure the search page for `walmart.com` is presented instead of the direct navigation to `walmart.com`

Test prioritisation of root URLs
1. Open **one tab** of [economist.com]( https://www.economist.com/)
2. Open **multiple tabs** of [economist.com/espresso](https://www.economist.com/espresso)
3. Open a new tab and type `e` into the address bar. 
4. Make sure [economist.com]( https://www.economist.com/) is the first top hit and [economist.com/espresso]( https://www.economist.com/espresso) is the second top hit

Test elimination of duplicates
1. Add [economist.com]( https://www.economist.com/) to your bookmarks
2. Visit [economist.com]( https://economist.com/), [www.economist.com]( https://www.economist.com/)
3. Type `economist.com` into the address bar
4. Make sure that just the first top hit links to [economist.com]( https://economist.com/)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
